ts/src/heap-profiler.ts

```diff
@@ -84,17 +83,17 @@ export function profile(
  * started with different parameters, this throws an error.
  *
  * @param intervalBytes - average number of bytes between samples.
- * @param stackDepth - maximum stack depth for samples collected.
+ * @param stackDepth - maximum stack depth for samples collected. This is currently no-op.
+ *                     Default stack depth of 128 will be used. Kept to avoid making breaking change.
  */
 export function start(intervalBytes: number, stackDepth: number) {
   if (enabled) {
     throw new Error(
-      `Heap profiler is already started  with intervalBytes ${heapIntervalBytes} and stackDepth ${stackDepth}`
+      `Heap profiler is already started  with intervalBytes ${heapIntervalBytes} and stackDepth 128`
     );
   }
   heapIntervalBytes = intervalBytes;
-  heapStackDepth = stackDepth;
-  startSamplingHeapProfiler(heapIntervalBytes, heapStackDepth);
+  startSamplingHeapProfiler(heapIntervalBytes, stackDepth);
```

Comment:
This function will return before the promise completes as it's fire and forget with this promise, might be why the test is failing